### ISSUE TITLE
_abnf.py/frame_buffer constructor is annotated incorrectly

### ISSUE THREAD

**Issue by jolaf** [websocket-client/websocket-client#948]
I observe the following crash under `typeguard`:
```
Traceback (most recent call last):
  File "test.py", line 750, in test
    webSocket = create_connection(WEBSOCKET_URL, sslopt = {"ca_certs": "/etc/ssl/certs/rootCA.crt"})
  File "/home/buildmaster/.local/lib/python3.10/site-packages/websocket/_core.py", line 603, in create_connection
    websock = class_(sockopt=sockopt, sslopt=sslopt,
  File "/home/buildmaster/.local/lib/python3.10/site-packages/websocket/_core.py", line 94, in __init__
    self.frame_buffer = frame_buffer(self._recv, skip_utf8_validation)
  File "/home/buildmaster/.local/lib/python3.10/site-packages/websocket/_abnf.py", line 276, in __init__
    def __init__(self, recv_fn: int, skip_utf8_validation: bool) -> None:
  File "/home/buildmaster/.local/lib/python3.10/site-packages/typeguard/_functions.py", line 138, in check_argument_types
    check_type_internal(value, annotation, memo)
  File "/home/buildmaster/.local/lib/python3.10/site-packages/typeguard/_checkers.py", line 766, in check_type_internal
    raise TypeCheckError(f"is not an instance of {qualified_name(origin_type)}")
typeguard.TypeCheckError: argument "recv_fn" (method) is not an instance of int
```
The problem is in `_abnf.py` line 276:
`def __init__(self, recv_fn: int, skip_utf8_validation: bool) -> None:`
As far as I could understand, `recv_fn` is expected to be a function returning `int`, not just `int` as specified.
Probably the correct annotation would be `Callable[[int], int]`.

**Comment by jolaf** [comment]
If the project has a test set, I recommend running it under `typeguard`, it would expose most incorrect annotations.

**Comment by engn33r** [comment]
This should be fixed by #949

The first goal to improve typing is to add mypy to CI and fix mypy errors. Then typeguard in the tests can come after
